### DISCUSSION TITLE
ignored build step monorepo dependencies

### DISCUSSION THREAD

**Discussion by BrianHung** [vercel/community#556] [Help]
https://vercel.com/docs/concepts/projects/overview#ignored-build-step

I have a monorepo project which includes backend code not deployed on Vercel in the same directory.

```
project
  apps/
    backend/
    frontend/
  packages/
    frontendDep1/
    frontendDep2/
    backendDep1/
  package.json
```

Because I have
```
Include source files outside of the Root Directory in the [Build Step](https://vercel.com/docs/deployments/build-step).
```
checked under `Root Directory`, every time the `backend` or `backendDepN` code changes, Vercel creates a new build despite not actually deploying it.

There is a way to limit new builds based on `git diff HEAD^ HEAD --quiet`, but that would require listing the `frontend` folder and its dependencies `frontendDepN` explicitly.

Is there an alternative where Vercel will only create a new build if new commits affect the frontend directory or its _ monorepo dependencies_ based on package.json or something? If it helps, I'm also using turborepo.

**Comment by samcx** [comment]
Have you tried taking a look at this Support Article → https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel?

**Comment by typeofweb** [comment]
I'm working on a package that does exactly that: [@typeofweb/ignore-monorepo-buildstep](https://www.npmjs.com/package/@typeofweb/ignore-monorepo-buildstep)
Usage is simple, just add `npx @typeofweb/ignore-monorepo-buildstep@latest $VERCEL_GIT_PREVIOUS_SHA` as your "Ignore Build Step" command in Vercel.
Currently, it only supports `pnpm` workspaces but I'm planning on adding more.

Here's the GitHub repo, any support, stars, PRs, issues… are welcome 😄  https://github.com/typeofweb/ignore-monorepo-buildstep

**Reply by amyegan** [reply]
Thanks for sharing your work! The new `VERCEL_GIT_PREVIOUS_SHA` [system environment variable](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables) is going to be very helpful for a lot of people. Awesome seeing it in use like this!
